src/services/statistic.ts

```diff
@@ -0,0 +1,24 @@
+import { Post } from '../../models/Posts';
+import { User } from '../../models/User';
+
+class StatisticService {
+  async getStatistic() {
+    try {
+      const usersIds = await User.find({}, { name: 1, surname: 1, _id: 1 }).lean();
+      const posts = await Post.find({}, { title: 1, user: 1, createdAt: 1 })
+        .sort({ createdAt: -1 })
+        .lean();
+      const usersWithPosts = usersIds.map((user) => {
+        // @ts-ignore
+        user.posts = posts.filter((post) => post.user.toString() === user._id.toString());
```

Comment:
Then you compare mongo ObjectId you should use equals on it.  
```suggestion
        user.posts = posts.filter((post) => post.user.equals(user._id));
```